### PR TITLE
add ssh native unix backend

### PR DESCRIPTION
**Pull request by geoalgo**

### CODE REVIEW

File: slurmpilot/remote_command.py [hunk 1]
```diff
@@ -95,6 +97,119 @@ def download_folder(self, remote_path: Path, local_path: Path):
...
+    def __init__(
+        self,
+        master: str,
+        user: str | None = None,
+        proxy: str | None = None,
+        local_dir: str | Path = None,
+    ):
+        """
+        Calls native unix ssh and scp with subprocesses. Should work reliably as long as ssh and scp works in a terminal
+        :param master:
```
Comment (aaronkl) [inline slurmpilot/remote_command.py:110]: missing description

File: slurmpilot/remote_command.py [hunk 2]
```diff
@@ -95,6 +97,119 @@ def download_folder(self, remote_path: Path, local_path: Path):
...
+            command=command, pty=pty, env=env, retries=retries
+        )
+
+    def _run_shell_command(
+        self, command: str, pty: bool = False, env: dict | None = None, retries: int = 0
+    ) -> CommandResult:
+        # TODO specify dir
+        with open(self.local_dir / "std.out", "w") as stdout:
+            with open(self.local_dir / "std.err", "w") as stderr:
+                process = subprocess.Popen(
```
Comment (aaronkl) [inline slurmpilot/remote_command.py:136]: any reason why you don't pass `env` to `Popen`?

File: slurmpilot/remote_command.py [hunk 3]
```diff
@@ -156,16 +271,19 @@ def run(
...
                 if fabric_result.failed:
                     if log_error:
-                        logging.debug(
+                        logging.info(
                             f"Command {command} failed\n{fabric_result.stderr}"
                         )
             except paramiko.ssh_exception.SSHException as e:
                 if log_error:
-                    logging.debug(
+                    print(traceback.format_exc())
```
Comment (aaronkl) [inline slurmpilot/remote_command.py:279]: debugging artifact?

File: slurmpilot/slurm_wrapper.py [hunk 4]
```diff
@@ -40,7 +41,23 @@ def __init__(
         clusters: List[str] | None = None,
         check_connection: bool = True,
         display_loaded_configuration: bool = False,
+        ssh_engine: str | None = None,
     ):
+        """
+        :param config:
```
Comment (aaronkl) [inline slurmpilot/slurm_wrapper.py:47]: missing description

**Review by aaronkl — APPROVED** [review]
LGTM. Just some small nitpicks

File: slurmpilot/remote_command.py (hunk 2 above)
Comment (geoalgo) [reply slurmpilot/remote_command.py:136]: In this case, we are only calling ssh and scp so I am not sure we need it (and if we need it better not to have it).

File: slurmpilot/remote_command.py (hunk 3 above)
Comment (geoalgo) [reply slurmpilot/remote_command.py:279]: ah you got me

**Pull request merged**